apis/appmesh/v1beta2/types.go

```diff
@@ -0,0 +1,194 @@
+package v1beta2
+
+// +kubebuilder:validation:Enum=http;tcp;http2;grpc
+type Protocol string
+
+const (
+	ProtocolHTTP  Protocol = "http"
+	ProtocolTCP   Protocol = "tcp"
+	ProtocolHTTP2 Protocol = "http2"
+	ProtocolGRPC  Protocol = "grpc"
+)
+
+// PortMapping refers to https://docs.aws.amazon.com/app-mesh/latest/APIReference/API_PortMapping.html
+type PortMapping struct {
+	// The port used for the port mapping.
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Maximum=65535
+	Port int64 `json:"port"`
+	// The protocol used for the port mapping.
+	Protocol Protocol `json:"protocol"`
```

Comment:
Can we add validation to this field as well to make sure we don't allow some random string?

Reply:
I see that 'Protocol' is already tagged with validation. Ignore my previous comment.

Reply:
:D yeah. the annotation on types works :D